src/coom.jl

```diff
@@ -30,39 +30,59 @@ julia> using TextAnalysis, DataStructures
   [1, 2]  =  2.0
   [3, 2]  =  0.3999
   [2, 3]  =  0.3999
+
+julia> using TextAnalysis, DataStructures
+       doc = StringDocument("This is a text about an apple. There are many texts about apples.")
+       docv = TextAnalysis.tokenize(language(doc), text(doc))
+       vocab = OrderedDict("This"=>1, "is"=>2, "apple."=>3)
+       TextAnalysis.coo_matrix(Float16, docv, vocab, 5, true, :directional)
+
+3×3 SparseArrays.SparseMatrixCSC{Float16,Int64} with 4 stored entries:
+  [2, 1]  =  1.0
+  [1, 2]  =  1.0
+  [3, 2]  =  0.1999
+  [2, 3]  =  0.1999
 ```
 """
 function coo_matrix(::Type{T},
-                    doc::Vector{<:AbstractString},
-                    vocab::OrderedDict{<:AbstractString, Int},
-                    window::Int,
-                    normalize::Bool=true) where T<:AbstractFloat
+    doc::Vector{<:AbstractString},
+    vocab::OrderedDict{<:AbstractString,
+        Int},
+    window::Int,
+    normalize::Bool=true,
+    mode::Symbol=:default) where {T<:AbstractFloat}
+    # Initializations
     n = length(vocab)
     m = length(doc)
     coom = spzeros(T, n, n)
     # Count co-occurrences
     for (i, token) in enumerate(doc)
+        inner_range = if mode == :directional
+            i:min(m, i + window)
+        else
+            max(1, i - window):min(m, i + window)
+        end
         row = get(vocab, token, nothing)
         isnothing(row) && continue
 
-        @inbounds for j in max(1, i - window):min(m, i + window)
+        # looking forward
+        @inbounds for j in inner_range
             i == j && continue
 
             wtoken = doc[j]
+            nm = T(ifelse(normalize, abs(i - j), 1))
```

Comment:
One more minor issue. Looks like this line can also be returned back after the check `isnothing(col) && continue` to do nothing when col is nothing.

Reply:
Oh, sure! Thanks!